core/indexer/process/transactions/grouper.go

```diff
@@ -0,0 +1,191 @@
+
+	for _, mb := range body.MiniBlocks {
+		mbHash, err := tg.calculateHash(mb)
+		if err != nil {
+			continue
+		}
+
+		mbStatus := computeStatus(tg.selfShardID, mb.ReceiverShardID)
+
+		switch mb.Type {
+		case block.TxBlock:
+			txs := getTransactions(txPool, mb.TxHashes)
+			for hash, tx := range txs {
+				dbTx := tg.txBuilder.buildTransaction(tx, []byte(hash), mbHash, mb, header, mbStatus)
+				addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, false)
+				if tg.shouldIndex(mb.ReceiverShardID) {
+					transactions[hash] = dbTx
+				}
+				delete(txPool, hash)
+			}
+		case block.InvalidBlock:
+			txs := getTransactions(txPool, mb.TxHashes)
+			for hash, tx := range txs {
+				dbTx := tg.txBuilder.buildTransaction(tx, []byte(hash), mbHash, mb, header, transaction.TxStatusInvalid.String())
+				addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, false)
+
+				dbTx.GasUsed = dbTx.GasLimit
+				fee := tg.txBuilder.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, dbTx.GasUsed)
+				dbTx.Fee = fee.String()
+
+				transactions[hash] = dbTx
+				delete(txPool, hash)
+			}
+		case block.RewardsBlock:
+			rTxs := getRewardsTransaction(txPool, mb.TxHashes)
+			for hash, rtx := range rTxs {
+				dbTx := tg.txBuilder.buildRewardTransaction(rtx, []byte(hash), mbHash, mb, header, mbStatus)
+				addToAlteredAddresses(dbTx, alteredAddresses, mb, tg.selfShardID, true)
+				if tg.shouldIndex(mb.ReceiverShardID) {
+					rewardsTxs = append(rewardsTxs, dbTx)
+				}
+				delete(txPool, hash)
+			}
+		default:
+			continue
+		}
+	}
+
+	return transactions, rewardsTxs
+}
+
+func (tg *txGrouper) shouldIndex(destinationShardID uint32) bool {
+	if !tg.isInImportMode {
+		return true
+	}
+
+	return tg.selfShardID == destinationShardID
+}
+
+func (tg *txGrouper) groupReceipts(header data.HeaderHandler, txPool map[string]data.TransactionHandler) []*types.Receipt {
+	receipts := make(map[string]*receipt.Receipt)
+	for hash, tx := range txPool {
+		rec, ok := tx.(*receipt.Receipt)
+		if !ok {
+			continue
+		}
+
+		receipts[hash] = rec
+		delete(txPool, hash)
+	}
+
+	dbReceipts := make([]*types.Receipt, 0)
+	for recHash, rec := range receipts {
+		dbReceipts = append(dbReceipts, tg.txBuilder.convertReceiptInDatabaseReceipt(recHash, rec, header))
+	}
+
+	return dbReceipts
+}
+
+func computeStatus(selfShardID uint32, receiverShardID uint32) string {
+	if selfShardID == receiverShardID {
+		return transaction.TxStatusSuccess.String()
+	}
+
+	return transaction.TxStatusPending.String()
+}
+
+func groupSmartContractResults(txPool map[string]data.TransactionHandler) map[string]*smartContractResult.SmartContractResult {
+	scResults := make(map[string]*smartContractResult.SmartContractResult)
+	for hash, tx := range txPool {
+		scResult, ok := tx.(*smartContractResult.SmartContractResult)
+		if !ok {
+			continue
+		}
+		scResults[hash] = scResult
+	}
+
+	return scResults
+}
+
+func getTransactions(txPool map[string]data.TransactionHandler,
```

Comment:
be consistent while naming. `groupSmartContractResults`, `getTransactions`....

Reply:
refactored.